dataservice/api/genomic_file/resources.py

```diff
@@ -0,0 +1,163 @@
+
+    @paginated
+    def get(self, after, limit):
+        """
+        Get paginated genomic_files
+
+        Retrieves the genomic files stored in the datamodel, then fetch
+        additional properties that are stored in indexd under the same uuid.
+        ---
+        template:
+          path:
+            get_list.yml
+          properties:
+            resource:
+              GenomicFile
+        """
+        # Get a page of the data from the model first
+        q = GenomicFile.query
+        refresh = True
+        while refresh:
+            refresh = False
+            pager = Pagination(q, after, limit)
+            for gf in pager.items:
+                merged = gf.merge_indexd()
+                if merged is None:
+                    refresh = True
+
+        return (GenomicFileSchema(many=True)
+                .jsonify(pager))
+
+    def post(self):
+        """
+        Create a new genomic_file
+        ---
+        template:
+          path:
+            new_resource.yml
+          properties:
+            resource:
+              GenomicFile
+        """
+        try:
+            gf = GenomicFileSchema(strict=True).load(request.json).data
+        except ValidationError as err:
+            abort(400,
+                  'could not create genomic_file: {}'.format(err.messages))
+
+        db.session.add(gf)
+        db.session.commit()
+
+        return GenomicFileSchema(
+            201, 'genomic_file {} created'.format(gf.kf_id)
+        ).jsonify(gf), 201
+
+
+class GenomicFileAPI(CRUDView):
+    """
+    GenomicFile API
+    """
+    endpoint = 'genomic_files'
+    rule = '/genomic-files/<string:kf_id>'
+    schemas = {'GenomicFile': GenomicFileSchema}
+
+    def get(self, kf_id):
+        """
+        Get a genomic file by id
+        ---
+        template:
+          path:
+            get_by_id.yml
+          properties:
+            resource:
+              GenomicFile
+        """
+        genomic_file = GenomicFile.query.get(kf_id)
+        if genomic_file is None:
+            abort(404, 'could not find {} `{}`'
+                  .format('genomic_file', kf_id))
+
+        # Merge will return None if the document wasnt found in indexd
+        merge = genomic_file.merge_indexd()
+        if merge is None:
+            abort(404, 'could not find {} `{}`'
+                  .format('genomic_file', kf_id))
+
+        sch = GenomicFileSchema(many=False)
+        return sch.jsonify(genomic_file)
+
+    def patch(self, kf_id):
+        """
+        Update an existing genomic_file
+        ---
+        template:
+          path:
+            update_by_id.yml
+          properties:
+            resource:
+              GenomicFile
+        """
+        body = request.json or {}
+        gf = GenomicFile.query.get(kf_id)
+        if gf is None:
+            abort(404, 'could not find {} `{}`'
+                  .format('genomic_file', kf_id))
+
+        # Fetch fields from indexd first
+        gf.merge_indexd()
```

Comment:
Should we return a 404 if the the merge results in a non-existent genomic file in indexd?